proposed/container-configuration-meta.md

```diff
@@ -29,10 +29,62 @@ The way a framework can automatically **discover the container definitions** of
+
+1. Ability to alias a container entry to another.
+
+1. Ability to modify an entry defined outside of the "module" before it is returned by the container. For instance: ability for a package providing a `Twig_Extension` to modify the `Twig_Environment` by calling `$environment->addExtension($extension)`, even if the `Twig_Environment` entry was not declared by the package itself.
+
+1. Ability to create container entries that are scalar values.
+
+1. Ability to create container entries that are constants (from the `define` keyword or the `const` keyword).
+
+1. Ability to create container entries that are numerically indexed arrays. Values of the array can be any valid container entry (i.e. objects, scalars, another array...)
 
-### 4.2. Features
+1. Ability to create container entries that are associative arrays (maps). Values of the array can be any valid container entry (i.e. objects, scalars, another array...)
+
+1. Ability for a package to extend those arrays (add elements to the arrays). Think about a package adding a PSR-7 middleware to the list of available middlewares.
+
+1. Ability for a package to manage priority in those arrays (add a service at the beginning, at the end, in the middle, or give a priority...).
+
+1. Ability to provide a default service that should be used when binding an interface.
+
+1. Ability to declare "lazy" services (services that are wrapped into proxy objects and instanciated only when needed).
+
+1. Ability to have several services for the same class or interface (for instance, several services representing a database connections)
+
+1. Ability to declare objects that are not services (for instance, ability to manage the menu of an application through `MenuItem` container entries that would be added to a `Menu` entry).
+
+1. Ability to have "optional" references: When binding a service to another service, this is the ability to bypass a reference if the entry does not exist in the container.
+
+1. Ability to have static tools analyzing the bindings (for instance, having Packagist analyze the bindings to search for some services...)
+
+1. Ability to have static tools help us edit the binding. For instance, a dedicated UI that can be used to create services and drag'n'drop services together (like Mouf does)
+
+1. Ability to perform simple computations on values before injecting them in a container entry (for instance, grab a return value of a function of a service, add it to another value and inject it in another service. This kind of feature comes "out of the box" for closure based services, and can be more complex to implement with definitions. See Symfony's "Expression language")
+
+When the list of features is complete, I propose we cast a vote on each feature, noting them from -10 (I think this feature is highly conterproductive, this is a showstopper for me), to 10 (I think this feature is absolutely needed and if it is not present, this is a show stopper for me).
```

Comment:
AFAIK "cast a poll" does not exist as expression. From a quick research on Google, this should be "create a poll" or "make a poll".